### DISCUSSION TITLE
my D435I depth camera mounted on arm_end_effector, how can i combined it with tesseract,to integrate octomap based on pointcloud topic in the planning server

### DISCUSSION THREAD

**Discussion by dbdxnuliba** [tesseract-robotics/tesseract_ros#197] [General]
and my D435i rgbd camera mounted on arm_end_effector, has been calibrated ,so I can publish the pointcloud topic,so I want an integrated octomap based on pointcloud topic in the planning server where the octomap is added during motion planning request, andy related work is appreciated

https://github.com/tesseract-robotics/tesseract_ros/issues/172

**Comment by Levi-Armstrong** [comment]
This can be accomplished currently by doing the following.

1. Get either the point cloud or the octree. 
2. Create `tesseract_geometry::Octree` type which can be constructed using either.
3. Create add link command, see this [example](https://github.com/tesseract-robotics/tesseract_planning/blob/0f0e26c1259f6c64340371198f869394d88c0d72/tesseract_examples/src/basic_cartesian_example.cpp#L62-L105).
4. Use the toMsg utility for converting the add link command to a message type and add this to your [planning request](https://github.com/tesseract-robotics/tesseract_ros/blob/master/tesseract_msgs/msg/PlanningRequest.msg) message.
